model.py

```diff
@@ -488,6 +488,10 @@ def to_dict(self, estimates, aggregate_dict=None):
         for k, target in enumerate(self.targets):
             estimates_dict[target] = estimates[:, k, ...]
 
+        # in the case of residual, we added another source
+        if k  < estimates.shape[1] - 1:
+            estimates_dict['residual'] = estimates[:, -1, ...]
+
```

Comment:
I hardcoded "residual" as a residual name

Reply:
maybe easier to just check for `self.residual` no?

Reply:
yes ok